Split TS rules for infantry, structures and vehicles

reaperrr opened the pull request:
Less scrolling, more structured.

reaperrr committed:
Split TS infantry.yaml into gdi-, nod- and player-infantry.yaml

reaperrr committed:
Rename civilian.yaml to civilian-structures.yaml

Move Sandbags to civilian structures.
Remove GAGREEN.

reaperrr committed:
Split TS vehicles.yaml into gdi-, nod- and player-vehicles.yaml

reaperrr committed:
structures.yaml split into faction-specific structure and defense yamls

reaperrr committed:
Improve rules yaml order in mod.yaml a little

Phrohdoh commented on mods/ts/mod.yaml:
```diff
@@ -61,17 +61,26 @@ Packages:
+	./mods/ts/rules/nod-infantry.yaml
+	./mods/ts/rules/nod-structures.yaml
+	./mods/ts/rules/nod-vehicles.yaml
+	./mods/ts/rules/player-infantry.yaml
```
How is this any different from `gdi-` and `nod-`? The name is also a bit misleading.

reaperrr replied:
```shared-``` sounds good, that's pretty self-explanatory at first glance.

reaperrr committed:
Rename yamls for actors available to both sides from player-* to shared-*

obrakmann commented on mods/ts/rules/civilian-structures.yaml:
```diff
@@ -1213,6 +1200,33 @@ GAOLDCC6:
+	Buildable:
+		Queue: Defense
+		BuildPaletteOrder: 1000
+		Prerequisites: ~disabled
```
This had `~structures.gdi` before. Move it to `gdi-defenses.yaml` and fix the prerequisite?

pchote replied:
This isn't supposed to be buildable by either faction.

obrakmann commented:
Looks good to me, :+1:

reaperrr committed:
Redistribute defense/support structure rules for more consistency

reaperrr commented:
Updated.
Renamed *-defenses to *-support and moved everything in the Defense queue into those files.

I didn't change the name of the queue itself for now since I suspect that to have a bigger chance of regressions, so I'd like to leave that for later.

abcdefg30 commented:
Needs a rebase.

reaperrr commented:
Rebased.

reaperrr committed:
Removes armament pistol from TS civilian default

Adds pistols to civ1 and civ3.

reaperrr commented:
Updated, removed pistol from civilian inf default and added it to civs 1 and 3 (civ2 has no shooting sequence).

abcdefg30 commented:
Shouldn't you also remove the sequences of `GAGREEN`?

abcdefg30 commented:
Looks otherwise ok to me and I didn't see anything ingame,
so :+1: / :white_check_mark: after above solved.

reaperrr committed:
Remove obsolete gagreen sequence

and removed a superflous newline

abcdefg30 commented:
Thanks. :)
